### ISSUE TITLE
Cloudwatch Logs sink does not handle non-existent log streams

### ISSUE THREAD

**Issue by mdfranz** [timberio/vector#586]
Events are only sent when the log stream exists

Tested in 0.3.0

```
TRACE source{name="logs"}:file-server: file_source::file_server: Continue watching file. path="/var/log/auth.log"              
thread 'tokio-runtime-worker-2' panicked at 'service must be discarded: CloudwatchError: No Streams Found', src/sinks/util/mod$
rs:171:27                                                                                                                      
TRACE source{name="logs"}:file-server: file_source::file_server: Continue watching file. path="/var/log/auth.log"              
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.                                        
stack backtrace:                                                                                                               
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace                                                                
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39                                                              
   1: std::sys_common::backtrace::_print                                                                                       
             at src/libstd/sys_common/backtrace.rs:71                                                                          
   2: std::panicking::default_hook::{{closure}}                                                                                
             at src/libstd/sys_common/backtrace.rs:59                                                                          
             at src/libstd/panicking.rs:197                                                                                    
   3: std::panicking::default_hook                                                                                             
             at src/libstd/panicking.rs:211                                                                                    
   4: std::panicking::rust_panic_with_hook                                                                                     
             at src/libstd/panicking.rs:474                                                                                    
   5: std::panicking::continue_panic_fmt                                                                                       
             at src/libstd/panicking.rs:381                                                                                    
   6: std::panicking::begin_panic_fmt                                                                                          
             at src/libstd/panicking.rs:336                                                                                    
   7: <vector::sinks::util::BatchServiceSink<T,S,B> as futures::sink::Sink>::start_send                                        
             at src/sinks/util/mod.rs:171                                                                                      
   8: vector::sinks::util::batch::BatchSink<B,S>::poll_send                                                                    
             at src/sinks/util/batch.rs:106                                                                                    
   9: <vector::sinks::util::batch::BatchSink<B,S> as futures::sink::Sink>::poll_complete                                       
             at src/sinks/util/batch.rs:182                                                                                    
  10: <alloc::boxed::Box<S> as futures::sink::Sink>::poll_complete                                                             
             at /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.27/src/sink/mod.rs:68                     
  11: <futures::stream::forward::Forward<T,U> as futures::future::Future>::poll                                                             at /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.27/src/stream/forward.rs:104
  12: <futures::future::map::Map<A,F> as futures::future::Future>::poll
             at /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-0.1.27/src/future/map.rs:30

SNIP

  74: tokio_threadpool::worker::Worker::do_run::{{closure}}
             at /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.14/src/worker/mod.rs:125
  75: std::thread::local::LocalKey<T>::try_with
             at /rustc/3c235d5600393dfe6c36eeed34042efad8d4f26e/src/libstd/thread/local.rs:299
  76: std::thread::local::LocalKey<T>::with
             at /rustc/3c235d5600393dfe6c36eeed34042efad8d4f26e/src/libstd/thread/local.rs:245
  77: tokio_threadpool::worker::Worker::do_run
             at /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.14/src/worker/mod.rs:116
  78: tokio_threadpool::pool::Pool::spawn_thread::{{closure}}
             at /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-threadpool-0.1.14/src/pool/mod.rs:344           

ERROR sink{name="my_aws_cloudwatch_logs_sink"}: vector::topology: Unhandled error
 INFO vector: Shutting down.
```

The behavior used by the AWS Python Logs agent is to automatically create new log streams so that these do not have to be configured on the backend with Terraform, Cloudformation, etc. 

pi@pi4b-7587c0da:/tmp $ cat vector.toml 

```
[sources.logs]   
   type = 'file'
   include = ['/var/log/auth.log']
   start_at_beginning = false
   ignore_older = 3600

[sinks.my_aws_cloudwatch_logs_sink]
  # REQUIRED - General
  type = "aws_cloudwatch_logs" 
  inputs = ["logs"]
  group_name = "/var/log/auth.log"
  region = "us-east-1"
  stream_name = "pi4b-7587c0da"
```

**Comment by LucioFranco** [comment]
@mdfranz Hi! Thanks for reporting this. Unfortunately, the way I originally took cloudwatch expected the streams to exist. Luckily, this approach has been completely revisited here https://github.com/timberio/vector/pull/519 which is currently sitting open but will dynamically create log streams and even let you extract stream names from event items. This should be merged into master early next week.

**Comment by mdfranz** [comment]
Thanks! https://github.com/timberio/vector/issues/476 captures these requirements as well.

**Comment by LucioFranco** [comment]
Closed via #519
